### DISCUSSION TITLE
Can I get inverse kinematic of 3dof robot in pinocchio ?

### DISCUSSION THREAD

**Discussion by lkx8421** [stack-of-tasks/pinocchio#2424] [Q&A] [answered]
In this [file](https://github.com/stack-of-tasks/pinocchio/blob/master/examples/inverse-kinematics.cpp) show how to use inverse kinematic of 6 dof robot, but how can I get 3dof robot joint with desire xyz position in pinocchio?

**Comment by jcarpent** [comment, marked as answer]
Yes, it is even more simpler than in this [example](https://github.com/stack-of-tasks/pinocchio/blob/master/examples/inverse-kinematics.cpp). You just need to take the linear part of the Jacobian associated with the translation and the linear part of the error. And that's all.

To help you, I've added one example both in C++ and Python.
After this [PR](https://github.com/stack-of-tasks/pinocchio/pull/2428) is merged, they will be added to Pinocchio's main repository.
